core/lib/src/fairing/ad_hoc.rs

```diff
@@ -1,8 +1,8 @@
 use std::sync::Mutex;
 
-use futures_util::future::BoxFuture;
+use futures_util::future::{Future, BoxFuture};
```

Comment:
I made this change and removed the direct dependency on `futures-util`, but I opted for `use futures::` instead of `use crate::futures::` based on length.

Reply:
Yeah, that's perfect.